### DISCUSSION TITLE
QR with Logo not working

### DISCUSSION THREAD

**Discussion by jasonwilliamsau** [chillerlan/php-qrcode#256] [Q&A] [answered]
I'm using the example here:
https://github.com/chillerlan/php-qrcode/blob/main/examples/imagickWithLogo.php
and set the pngLogo option
`$options->pngLogo = '/Users/mbp/website/public_html/g/1/32/bg-icon-312x312.png';`

it generates an png with the code but the logo space is blank.
![Screenshot 2024-04-19 at 2 33 12 PM](https://github.com/chillerlan/php-qrcode/assets/16073729/5a226c3a-c086-4bdb-a90a-8d2439041580)

Further to this after more testing:
Using the GD example will render the logo correctly so the issue appears to be imagick related. 

However, trying the Imagick Background image example
https://github.com/chillerlan/php-qrcode/blob/main/examples/imagickImageAsBackground.php
it works.
![Screenshot 2024-04-19 at 3 53 58 PM](https://github.com/chillerlan/php-qrcode/assets/16073729/0fdcf7ed-e397-4355-be7a-6569455f3cb2)

Does anyone have any suggestions?

**Comment by codemasher** [comment]
Hey, i just tested the `imagickWithLogo` example and for some reason i don't get a logo either, so i can confirm the issue is not on your end. Sorry for the inconvenience!

**Reply by codemasher** [reply]
One question: which version of `php-qrcode` are you running? You linked the example from the dev-main branch.

**Reply by jasonwilliamsau** [reply]
no probs... i've been testing this all day,... 

I just found if I set `$options->imageTransparent    = true;` to false, it worked.

Currently doing tests with
```
$options->bgColor             = '#fefefe';
$options->imageTransparent    = true;
$options->transparencyColor   = '#fefeff'; 
```

**Reply by jasonwilliamsau** [reply]
yep - `dev-main`.
I had earlier issues as composer was using the 5x branch. 
I forced it to main and it solved some other issues i was getting

**Reply by codemasher** [reply]
Yea, the issue with v5.x and examples from the main branch is known, see https://github.com/chillerlan/php-qrcode/commit/900c8aea4f233cd8f3d3047de8978b987aa0a317

**Reply by codemasher** [reply, marked as answer]
I found the issue - the mode should be `Imagick::COMPOSITE_BLEND` here:

https://github.com/chillerlan/php-qrcode/blob/181e12c70c90f27be928f55067b4861d988c5c7c/examples/imagickWithLogo.php#L49
